lib/posix/options/clock.c

```diff
@@ -43,6 +43,24 @@ int z_impl___posix_clock_get_base(clockid_t clock_id, struct timespec *base)
 			*base = rt_clock_base;
 		}
 		break;
+#if defined(CONFIG_SCHED_THREAD_USAGE_ANALYSIS) && defined(CONFIG_POSIX_THREAD_CPUTIME)
+	case CLOCK_THREAD_CPUTIME_ID:
+		K_SPINLOCK(&rt_clock_base_lock) {
+			uint64_t ns;
+			k_thread_runtime_stats_t stats;
+			k_tid_t thread = k_current_get();
+
+			if (!k_thread_runtime_stats_is_enabled(thread)) {
+				errno = EINVAL;
+				return -1;
+			}
+			k_thread_runtime_stats_get(thread, &stats);
+			ns = k_cyc_to_ns_floor64(stats.execution_cycles);
+			base->tv_sec = ns / NSEC_PER_SEC;
+			base->tv_nsec = ns % NSEC_PER_SEC;
+		}
+		break;
+#endif /* CONFIG_SCHED_THREAD_USAGE_ANALYSIS */
```

Comment:
nit: this doesn't match L46 & L47

Reply:
Yes, I noticed it. But I'm wondering if I should change it to `CONFIG_SCHED_THREAD_USAGE_ANALYSIS && CONFIG_POSIX_THREAD_CPUTIME`, or ignore it.

Reply:
I'd change it to `CONFIG_SCHED_THREAD_USAGE_ANALYSIS && CONFIG_POSIX_THREAD_CPUTIME`

Reply:
I've change, Please to see :)